packages/flutter/lib/src/material/tabs.dart

```diff
@@ -174,17 +174,50 @@ class _TabStyle extends AnimatedWidget {
 
   final TextStyle? labelStyle;
   final TextStyle? unselectedLabelStyle;
-  final bool selected;
+  final bool isSelected;
   final Color? labelColor;
   final Color? unselectedLabelColor;
   final Widget child;
 
+  MaterialStateColor _resolveWithLabelColor(BuildContext context) {
+    final ThemeData themeData = Theme.of(context);
+    final TabBarTheme tabBarTheme = TabBarTheme.of(context);
+    final Animation<double> animation = listenable as Animation<double>;
+
+    final Color selectedColor = labelColor
+        ?? labelStyle?.color
+        ?? tabBarTheme.labelColor
+        ?? tabBarTheme.labelStyle?.color
+        ?? themeData.primaryTextTheme.bodyLarge!.color!;
+
+    if (selectedColor is MaterialStateColor) {
```

Comment:
This is good, in that it supports a MaterialStateColor from the widget or the theme, however it doesn't support lerping between the selected and unselected colors if the labelColor is specified as a MaterialStateColor. This could be corrected here, by resolving the MaterialState selectedColor against `<MaterialState>{MaterialState.selected}` and `<MaterialState>{}` and using those two colors in the resolveWith expression below.

Reply:
Oh right, thanks for catching that. will add a test for it too.